api/catalog/api/serializers/audio_serializers.py

```diff
@@ -171,6 +173,15 @@ class AudioSerializer(MediaSerializer):
         help_text="A link to an endpoint that provides similar audio files.",
     )
 
+    # Add-on data
+    peaks = serializers.SerializerMethodField()
+
+    @staticmethod
+    def get_peaks(obj):
+        if isinstance(obj, Hit):
+            obj = Audio.objects.get(identifier=obj.identifier)
+        return obj.get_or_create_waveform(short_circuit=True)
```

Comment:
This will still cause n+1 query execution, specifically it will create `n` `create` queries.

Reply:
There are potentially N database queries, yes, but those are on the primary key (indexed) and are considerably less expensive than N queries from the frontend to the API over HTTP.

Reply:
> are considerably less expensive than N queries from the frontend to the API over HTTP

Is the concern here the HTTP responses? 20 extra HTTP requests per search request is pretty trivial compared to what Django and gunicorn can handle as far as I understand, but I agree that removing the need for them is better. However, we already make this number of requests for thumbnails anyway; if we're doing this kind of optimization we should reconsider how thumbnails are requested as well.

Likewise, the HTTP requests to the API do use the primary key to access the add on, the audio object identifier on the AddOn table is the PK for that table.

Am I misunderstanding the way that this PR makes is situation better aside from removing the additional HTTP requests though?

Reply:
No @sarayourfriend your understanding is correct that the PR only removes 20 additional HTTP requests. It doesn't improve anything else.

Reply:
Okay. I don't feel comfortable approving this approach as it actually adds an actual n+1 query problem to the audio results endpoint. In fact we're creating the objects in a loop that will block each iteration on the database interaction.

If others are fine with this then feel free to go ahead with it but please open an issue to address the n+1 query in the audio results view.